tensorflow_recommenders_addons/dynamic_embedding/core/lib/nvhash/cudf/concurrent_unordered_map.cuh

```diff
@@ -889,7 +949,107 @@ public:
 
         return iterator( m_hashtbl_values,m_hashtbl_values+hashtbl_size, current_hash_bucket);
     }
-    
+                               
+    template<typename hash_value_type = typename Hasher::result_type>
+    __forceinline__
+    __device__ iterator accum(const key_type& key, const mapped_type& vals, const bool &exist, 
+                               bool precomputed_hash = false,
+                               hash_value_type precomputed_hash_value = 0)
+    {
+        const size_type hashtbl_size    = m_hashtbl_size;
+        value_type* hashtbl_values      = m_hashtbl_values;
+
+        hash_value_type hash_value{0};
+
+        // If a precomputed hash value has been passed in, then use it to determine
+        // the write location of the new key
+        if(true == precomputed_hash)
+        {
+          hash_value = precomputed_hash_value;
+        }
+        // Otherwise, compute the hash value from the new key
+        else
+        {
+          hash_value = m_hf(key);
+        }
+
+        size_type current_index         = hash_value % hashtbl_size;
+        value_type *current_hash_bucket = &(hashtbl_values[current_index]);
+
+        const key_type insert_key = key;
```

Comment:
It will make a copy of `key`. Better use key directly or use `insert_key` as reference.